### ISSUE TITLE
`TokenClassifier` objects return wrong neural type

### ISSUE THREAD

**Issue by PeganovAnton** [NVIDIA/NeMo#1273]
**Describe the bug**

The `TokenClassifier` objects return a tensor of neural type `LogitsType`, whereas if the attribute `log_softmax` is `True`, the module return type has to have `LogprobsType` neural type. This inconsistency leads to the inconsistent input type and name of the parameter `logits` of the `SmoothedCrossEntropyLoss.forward` method.

**Comment by ericharper** [comment]
This should be fixed now: https://github.com/NVIDIA/NeMo/pull/1405

Please let us know if there are still issues.
